pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -78,9 +78,9 @@ type EC2NodeClassSpec struct {
 	InstanceProfile *string `json:"instanceProfile,omitempty"`
 	// Tags to be applied on ec2 resources like instances and launch templates.
 	// +kubebuilder:validation:XValidation:message="empty tag keys aren't supported",rule="self.all(k, k != '')"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching eks:eks-cluster-name",rule="self.all(k, k !='eks:eks-cluster-name')"
```

Comment:
Should we change this logic in beta? I think this should only apply to the new API